Sming/Components/crypto/include/Crypto/Blake2s.h

```diff
@@ -0,0 +1,71 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * blake2s.h
+ *
+ ****/
+
+#pragma once
+
+#include <sming_attr.h>
+#include <stdint.h>
+#include <cstring>
+#include <cassert>
+#include "Blob.h"
+
+namespace Crypto
+{
+class Blake2sEngine
+{
+public:
+	static constexpr const char* name = "blake2s";
+	static constexpr size_t hashsize = 32; // Maximum hash size of Blake2s. A shorter Hash can be selected at runtime.
+	static constexpr size_t blocksize = 64;
+
+	static constexpr size_t maxkeysize = 32;
+
+	void init(const Secret& key, size_t hashSize = hashsize);
+	void init(size_t hashSize = hashsize)
+	{
+		initCommon(hashSize, 0);
+	}
+
+	void update(const void* data, size_t size);
+	void final(uint8_t* hash);
+
+	uint64_t get_state(void* stateOut)
+	{
+		memcpy(stateOut, &state, sizeof(state));
+		return state.t64;
+	}
+
+	void set_state(const void* stateIn, uint64_t count)
+	{
+		memcpy(&state, stateIn, sizeof(state));
+		assert(state.t64 == count);
+	}
```

Comment:
I've do admit I wondered what these methods are for.

Since copying only part of the state would be even more confusing for anyone stumbling over this, I'll stub these methods.